Sources/App/Views/PackageController/PackageShow+View.swift

```diff
@@ -256,6 +268,16 @@ enum PackageShow {
                 )
             )
         }
+        func sidebarPublishedAndUpdatedSection() -> Node<HTML.BodyContext> {
+            .unwrap(packageSchema) {
+                .section(
+                    .class("sidebar_package_published"),
+                    .div(.small(.text("Published \(Self.dateFormatter.string(from:$0.dateCreated!))"))),
```

Comment:
Force unwrap is no good.

Reply:
```swift
                    .unwrap($0.datePublished, {
                        .div(.small(.text("Published \(Self.dateFormatter.string(from: $0))")))
                    }),
                    .unwrap($0.dateModified, {
                        .div(.small(.text("Last updated: \(Self.dateFormatter.string(from: $0))")))
                    })
```

should work but I think we can actually make those dates non-optional in the model and bail in `init?` should they be nil (which is practise they wouldn't):

```swift
        init?(result: PackageController.PackageResult) {
            let package = result.package
            let repository = result.repository
            guard
                let repositoryOwner = repository.owner,
                let repositoryName = repository.name,
                let datePublished = repository.firstCommitDate,
                let dateModified = repository.lastCommitDate,
                let dateLastActive = repository.lastActivityAt
            else {
                return nil
            }
            
            self.init(
                repositoryOwner: repositoryOwner,
                repositoryName: repositoryName,
                organisationName: repository.ownerName,
                summary: repository.summary,
                licenseUrl: repository.licenseUrl,
                version: PackageShow.releaseInfo(
                    packageUrl: package.url,
                    defaultBranchVersion: result.defaultBranchVersion,
                    releaseVersion: result.releaseVersion,
                    preReleaseVersion: result.preReleaseVersion).stable?.link.label,
                repositoryUrl: package.url.droppingGitExtension,
                datePublished: datePublished,
                dateModified: dateModified,
                dateLastActive: dateLastActive,
                keywords: repository.keywords
            )
        }
```

Reply:
(Throwing `dateLastActive` in there as well for symmetry.)

Reply:
Actually, on that note - `dateLastActive` doesn't seem to be used anywhere. My understanding of #1640 was that this should replace `dateModified`?

So I think we should drop that property and instead assign `lastActivityAt` to `dateModified` instead of using `lastCommitDate`.

But I'll let @daveverwer confirm that - I just wanted to give it a pass because I know Dave isn't around today.

Reply:
I've gone ahead and tried the `init?` change that I proposed. It can be made to work but it means we have to update some unrelated tests, because we're now more strict about populating `firstCommitDate` and `lastCommitDate` in our mocks.

I'll add a PR against this PR for discussion.

Reply:
Yea I think it's extremely unlikely that either of these fields will ever be nil but if they are it's fine to omit it with the `unwrap`.

Reply:
In terms of the markup here, the `section` tag is a good idea, but let's replace the `div` tag with a `p` so it gives a little padding above the footer. I'd also keep both fields on one line, maybe separated with a dash, and keep the formatting with the colon consistent. I'd also probably put a `hr` tag in there to separate it from the bottom of the README content.

Something like this:

<img width="877" alt="Screenshot 2022-04-19 at 08 41 31@2x" src="https://user-images.githubusercontent.com/5180/163951485-184ceec1-bb23-4353-bd54-c7bd38dff30c.png">

<img width="487" alt="Screenshot 2022-04-19 at 08 42 29@2x" src="https://user-images.githubusercontent.com/5180/163951647-317ed9d2-908c-4688-8575-72a5b1c9a26f.png">

I think we'll also get away without any section specific CSS, so the class can be omitted too.